pravega-sensor-collector/src/main/java/io/pravega/sensor/collector/file/LogFileSequenceProcessor.java

```diff
@@ -114,21 +106,59 @@ public void processNewFiles() throws Exception {
 
     protected void findAndRecordNewFiles() throws Exception {
         final List<FileNameWithOffset> directoryListing = getDirectoryListing();
-        final List<FileNameWithOffset> completedFiles = state.getCompletedFiles();
+        final List<FileNameWithOffset> completedFiles = state.getCompletedFileRecords();
         final List<FileNameWithOffset> newFiles = getNewFiles(directoryListing, completedFiles);
-        state.addPendingFiles(newFiles);
+        state.addPendingFileRecords(newFiles);
     }
 
     /**
      * @return list of file name and file size in bytes
      */
     protected List<FileNameWithOffset> getDirectoryListing() throws IOException {
         log.info("getDirectoryListing: fileSpec={}", config.fileSpec);
-        final List<FileNameWithOffset> directoryListing = FileUtils.getDirectoryListing(config.fileSpec, config.fileExtension);
+        final List<FileNameWithOffset> directoryListing = getDirectoryListing(config.fileSpec, config.fileExtension);
         log.trace("getDirectoryListing: directoryListing={}", directoryListing);
         return directoryListing;
     }
 
+    /**
+     * @return list of file name and file size in bytes
+     * Handle the below cases
+     *  1. If given file path does not exist then log the message and continue
+     *  2. If directory does not exist and no file with given extn like .csv then log the message and continue
+     *  3. check for empty file, log the message and continue with valid files
+     *
+     */
+    static protected List<FileNameWithOffset> getDirectoryListing(String fileSpec, String fileExtension) throws IOException {
```

Comment:
Please doublecheck whether this change is compatible with https://github.com/pravega/pravega-sensor-collector/pull/16